### ISSUE TITLE
Weather data EDA

### ISSUE THREAD

**Issue by andrew3cooper** [rsangole/capstone_project#21]
There is a lot of weather data available and a lot of missing data.  So far, I just pulled in a few variables (precipitation and temperature) from two stations (O'Hare and Midway airport weather stations).  We could use additional EDA to see which other weather variables from other stations are populated enough to use and look like they may be useful.  Alternatively, we could run a distance matrix or something to find the closest weather station to each trap (but first limit to stations with nearly-complete daily data!).

**Comment by rsangole** [comment]
What I have in mind:

Step 1 - Use the Midway and O'Hare airport data as a start point. Include a haversine distance function to use the "nearest" weather station data while modeling

Step 2 - For the final few "best" models, try to incorporate zipcode-level hyper-local weather data and see if things improve.

Thoughts?

I'm sure from the visualizers / dashboarders standpoint, we can leverage these data too.

**Comment by andrew3cooper** [comment]
Agreed.

**Comment by andrew3cooper** [comment]
Rahul, I took another stab at weather data since we have partial data from >300 weather stations in the surrounding area.  I built off what you did with Haversine distances to make a distance matrix between all weather stations and all trap locations.  I then limited to stations with nearly-complete time series data for precipitation (there are only about 10 stations that don't have a lot of missing data).  I can clean this up tomorrow, but the purpose would to have a single new time series column for precipitation at nearest weather station.  I did the same thing for temperature but I'm not satisfied with the data.  I'm going to take one more look at other weather variables we might want to include.  I saw that counts of "heating days" and "cooling days" were predictive in another WNV mosquito analysis....
